spec/services/hyrax/collections/migration_service_spec.rb

```diff
@@ -277,4 +446,26 @@ def expect_no_access(permission_template_id, agent_type, access, agent_ids)
       expect(pta).to be_empty
     end
   end
+
+  def expect_solr_access(permission_template_id, pt_agent_type, pt_access, solrdoc_access)
+    ptas = Hyrax::PermissionTemplateAccess.where(permission_template_id: permission_template_id, agent_type: pt_agent_type, access: pt_access)
+    expect_solr_group_access(ptas, solrdoc_access) if pt_agent_type == 'group'
+    expect_solr_user_access(ptas, solrdoc_access) if pt_agent_type == 'user'
+  end
+
+  def expect_solr_group_access(permission_templates, solrdoc_access)
+    permission_templates.each do |pta|
+      if ['registered', 'public'].include? pta.agent_id
+        expect(solrdoc_access).not_to include pta.agent_id
```

Comment:
I wasn't expecting to find the negative expectations here.  I would have thought they would be separate and distinct from the positive expectations.  I think they might be more obvious if kept outside of `expect_solr_access` and maybe in a similarly named `expect_no_solr_access`?

Reply:
These are not really testing anything specific to the migration_service.  The conditions expressed here are tested in...
* spec/services/hyrax/collections/permissions_service_spec.rb.source_ids_for_deposit returns admin set ids where user has deposit access except excluded groups